lib/tx_build/transaction.ex

```diff
@@ -77,6 +83,46 @@ defmodule Stellar.TxBuild.Transaction do
     )
   end
 
+  @spec hash(tx :: t()) :: String.t()
```

Comment:
After reviewing other SDK implementations, your recommendation makes more sense because **TxBuild** is the base of the transaction. I'll make the change.